### DISCUSSION TITLE
Automatic history cleanup?

### DISCUSSION THREAD

**Discussion by rvendrame** [hassio-addons/addon-traccar#76] [Ideas]
New to both traccar and Home assistant, so apologize if this is already implemented. 

Traccar is running perfectly for a week in my HAl.  This includes remote access, port forwarding and all nice features.  Congrats for the excellent add-on!  

https://www.traccar.org/clear-history/

Is it possible to activate Database + Log cleanup as a periodic task in Traccar's add-on, as per the link above?  

Maybe a parameter in the add-on could provide the number of days to keep in log files / database tracking history ( zero = forever ).  Feasible? 

I'm not expert/brave enough to change things inside the containers (yet).  

Thanks!

**Comment by frenck** [comment]
no way, that is how traccar solves it? 😢  That is so ugly to put that on the end-user. I guess it saves them the need to build it into the application.

I don't see how to add that to the add-on correctly. Besides, the database for the add-on is external (other add-on).

**Reply by rvendrame** [reply]
Hi @frenck,  yes, I agree it is ugly.  That's why I asked if something would be done in the add-on ;-) 

I tried to search Traccar forum --- But there is no search option there ?!?!?!?! 

Anyway I found this in the Traccar [website](https://www.traccar.org/configuration-file/) 

<img width="895" alt="image" src="https://user-images.githubusercontent.com/6721878/111169705-f7d2e100-8581-11eb-9fe7-787b980468e2.png">

So I added this line in my /config/traccar.xml : 

`    <entry key='database.historyDays'>1</entry>`

And restarted Traccar add-on;  will see the results.  

In case the above does the trick for Db history --- Would it be possible to implement the log rotation part in the add-on? 

find /opt/traccar/logs/ -mtime +3 -type f -delete

? 
Thanks!

**Reply by frenck** [reply]
Logs are not persistent.

**Reply by rvendrame** [reply]
Just had information that the configuration option above was 'removed' :-(   Will try to adopt the scripts to run on my Maria DB.  That should be enough --- As I understood from your comments that Traccar logs are no eating disk.  Thanks anyway.
